### PR TITLE
Chisel compat

### PR DESCRIPTION
**Pull request by WaitingIdly**
what this PR adds:
- add sound bracket handler
- fix typo in `extendedcraft.groovy` example file (was called `extremecrafting`)
- add chisel compat for creating or removing groups and adding or removing items from groups
- add error handling for oredict groups (cannot remove items from the chisel variation, have to remove the items from the oredict)
- an examples file for the above

### CODE REVIEW

File: src/main/java/com/cleanroommc/groovyscript/compat/mods/chisel/Carving.java [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+        public CarvingGroup add(ItemStack... items) {
+            for (ItemStack item : items) {
+                add(item);
+            }
+            return this;
+        }
+
+        public CarvingGroup add(Collection<ItemStack> items) {
+            for (ItemStack item : items) {
```
Comment (Testure) [inline src/main/java/com/cleanroommc/groovyscript/compat/mods/chisel/Carving.java:141]: `items.forEach(this::add)` can be used here

File: src/main/java/com/cleanroommc/groovyscript/compat/mods/chisel/Carving.java [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+        this.groupScripted = new ArrayList<>();
+    }
+
+    public static CarvingGroup carvingGroup(String group) {
+        return new CarvingGroup(group);
+    }
+
+    @Override
+    public void onReload() {
+        removeScripted().forEach(pair -> CarvingUtils.getChiselRegistry().removeVariation(pair.getValue(), pair.getKey()));
```
Comment (Testure) [inline src/main/java/com/cleanroommc/groovyscript/compat/mods/chisel/Carving.java:35]: Could you cache the `getChiselRegistry()` result? (And make sure it's not null)

File: src/main/java/com/cleanroommc/groovyscript/compat/mods/chisel/Carving.java [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+            GroovyLog.msg("Error removing a Chisel Carving variation")
+                    .add("you cannot remove variations to Oredict chisel groups {}", groupName)
+                    .add("instead, edit the oredict via `oredict.remove('{}', {})`", groupName, IngredientHelper.asGroovyCode(item, false))
+                    .error()
+                    .post();
+        }
+    }
+
+    public void setSound(String group, SoundEvent sound) {
+        setSound(CarvingUtils.getChiselRegistry().getGroup(group), sound);
```
Comment (Testure) [inline src/main/java/com/cleanroommc/groovyscript/compat/mods/chisel/Carving.java:74]: Null check on `getGroup()`?

**Review by Testure — CHANGES_REQUESTED** [review]
You turned my scroll bar yellow with NPE warnings. Other than that, looks good.

File: src/main/java/com/cleanroommc/groovyscript/compat/mods/chisel/Carving.java (hunk 1 above)
Comment (WaitingIdly) [reply src/main/java/com/cleanroommc/groovyscript/compat/mods/chisel/Carving.java:141]: its whats used in `AbstractRecipeBuilder`, so i would prefer not to change this

File: src/main/java/com/cleanroommc/groovyscript/compat/mods/chisel/Carving.java [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+import team.chisel.api.carving.CarvingUtils;
+import team.chisel.api.carving.ICarvingGroup;
+import team.chisel.api.carving.ICarvingRegistry;
+
+import java.util.ArrayList;
+import java.util.Collection;
+
+public class Carving extends VirtualizedRegistry<Pair<String, ItemStack>> {
+
+    private static final ICarvingRegistry registry = CarvingUtils.getChiselRegistry();
```
Comment (Testure) [inline src/main/java/com/cleanroommc/groovyscript/compat/mods/chisel/Carving.java:19]: Can you reassign `registry` later on such as on reload or wherever in the loading process chisel would be ready?
`static final` assignment means if the first attempt returns null, the API becomes unusable.

**Review by Testure — CHANGES_REQUESTED** [review]

File: src/main/java/com/cleanroommc/groovyscript/compat/mods/chisel/Chisel.java [hunk 5]
```diff
@@ -0,0 +1,13 @@
+package com.cleanroommc.groovyscript.compat.mods.chisel;
+
+import com.cleanroommc.groovyscript.compat.mods.ModPropertyContainer;
+
+public class Chisel extends ModPropertyContainer {
+
+    Carving carving = new Carving();
```
Comment (brachy84) [inline src/main/java/com/cleanroommc/groovyscript/compat/mods/chisel/Chisel.java:7]: `public final`

**Review by brachy84 — CHANGES_REQUESTED** [review]

File: src/main/java/com/cleanroommc/groovyscript/compat/mods/chisel/Carving.java (hunk 4 above)
Comment (brachy84) [reply src/main/java/com/cleanroommc/groovyscript/compat/mods/chisel/Carving.java:19]: Just use `CarvingUtils.getChiselRegistry()` everywhere

**Review by brachy84 — APPROVED** [review]

**Pull request merged**